client/Rating&Reviews/RatingsAndReviews.jsx

```diff
@@ -33,6 +54,108 @@ class RatingsAndReviews extends React.Component {
     this.retrieveAllReviews(productId, sort, totalReviews);
   }
 
+  handleSearch(event) {
+    const {filteredReviews} = this.state;
+    const searchTerm = event.target.value;
+    const searchedReviews = [];
+    // const regex = new RegExp(searchTerm);
+
+    if (searchTerm.length >= 3) {
+      for (let i = 0; i < filteredReviews.length; i++) {
+        if (filteredReviews[i].summary && filteredReviews[i].summary.includes(searchTerm)) {
+          searchedReviews.push(filteredReviews[i]);
+        } else if (filteredReviews[i].body.includes(searchTerm)) {
+          searchedReviews.push(filteredReviews[i]);
+        }
+      }
+
+      // const newReview = {};
+      // newReview = Object.assign(newReview, filteredReviews[i]);
+
+      // if (filteredReviews[i].summary && filteredReviews[i].summary.includes(searchTerm)) {
+      //   found = true;
+      //   const newSummary = filteredReviews[i].summary.split(regex);
+      //   for (let j = 1; j < newSummary.length; j += 2) {
+      //     newSummary[j] = <mark key={j}>{searchTerm}</mark>;
+      //   }
+      //   newReview.summary = <>{newSummary}</>
+      // }
+
+      // if (filteredReviews[i].body.includes(searchTerm)) {
+      //   found = true;
+      //   const newBody = filteredReviews[i].body.split(regex);
+      //   for (let j = 1; j < newBody.length; j += 2) {
+      //     newBody[j] = <mark key={j}>{searchTerm}</mark>;
+      //   }
+      //   newReview.body = <>{newBody}</>
+      // }
+
+      // if (found) {
+      //   searchedReviews.push(newReview);
+      // }
+
+      this.setState({
+        searchTerm,
+        searchedReviews
+      });
+    } else {
+      this.setState({
+        searchTerm,
+        searchedReviews: []
+      })
+    }
+  }
+
+  handleFilter(rating) {
+    const {filters, reviews} = this.state;
+    const newFilters = [...filters];
+
+    if (rating === 0) {
+      this.setState({
+        filters: []
+      }, () => {
+        this.setState({
+          filteredReviews: this.filterReviews(reviews)
+        });
+      });
+    } else if (!newFilters.includes(rating)) {
```

Comment:
Great work Kyle, you may consider using some conditions that could trim up your if/else blocks.